site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,177 @@
+
+{% instruction 'Add the following imports to the `main` module.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' lines:2-3 %}
+
+{% instruction 'Now, add the following after the `applicationContext` declaration.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:31 %}
+
+This statement does two things. First, it uses the `Injector` factory function to wrap the `applicationContext` with the `BaseInjector`, a lightweight component that allows any object that extends `Evented` to be injected via a `Container`, which we will talk about in the next section. This statement also registers the `Injector` with the application's central registry. The registry provides a way to register a widget via a label, making it accessible to other parts of the application. You can learn more in the [registry tutorial](../comingsoon.html).
+
+Now that the `Injector` is defined and registered, it is time to create the components that will use it. In the next section, we will create a special type of non-visual widget called a `Container` that will allow state to be injected into the `WorkerForm` and `WorkerContainer` widgets.
+
+{% section %}
+
+## Creating state containers
+
+{% task 'Create `Containers` that will allow state to be injected into widgets' %}
+
+On their own, `Injectors` are not able to help us very much because widgets expect state to be pushed into them (via their properties), but state must be pulled from an `Injector`. `Containers` are designed to coordinate the injection - they pull state from injectors and push it into widgets.
+
+Normally, a separate `Container` is created for each widget that needs to have state injected. In the demo application, we have two widgets that rely on application state - `WorkerContainer` and `WorkerForm`. Let's start with the `WorkerContainer`. `Containers` are given the same name as the widget that they contain with the suffix `Container` appended. To keep things organized, they are also stored in a different sub-package - `containers`.
+
+{% instruction 'Add the following imports to the `WorkerContainerContainer` in the `containers` sub-package' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerContainerContainer.ts' lines:1-4 %}
+
+The first import gives the module access to the `Container` factory function which will be used to construct the container. The other imports allow the module to use the `ApplicationContext` to extract state, use the `WorkerContainerProperties` to receive properties from its parent, and the `WorkerContainer` class that will be wrapped by the container.
+
+Next, we need to address the fact that the container has two places to get properties from - its parent widget and the `ApplicationContext`. To tell the container how to manage this, we will create a function called `getProperties`.
+
+{% instruction 'Add the `getProperties` function to the `WorkerContainerContainer` module.' %}
```

Comment:
Having something called WorkerContainerContainer tells me we should have picked a better name for WorkerContainer.